presto-main/src/main/java/com/facebook/presto/event/QueryMonitor.java

```diff
@@ -512,4 +545,22 @@ private static StageCpuDistribution computeCpuDistribution(StageInfo stageInfo)
                 (long) snapshot.getTotal(),
                 snapshot.getTotal() / snapshot.getCount());
     }
+
+    public static class JsonPlanFragment
```

Comment:
I am curious whether you considered simply having a list rather than a map? I think the list being ordered makes this field a lot more readable and consistent. You could also look up things just as easily list[stageNumber] == map[stageNumber]. It would also remove the need to construct a map. Sorry for the back and forth, but I'm getting new ideas as I look at this repeatedly.

Reply:
A list relies on the elements being sorted by fragment id. 
1. I'm not even sure that it currently is, and I wouldn't rely on since it's not in the method contract so we'd need to add sorting here.  That means it wouldn't simplify the code to use a list. 
2. Even if we sorted it here, it's more ambiguous for the user who just has the json.  I'd rather have a map where the mapping so that the fragment mapping is explicit.  

What about a sorted map?  Also, could we rely on the PlanFragmentIds being integers for the ordering (even though they're technically strings)?

Reply:
Not a strong view, but I think sorting and using a list would actually be more concise. I can see it being like 3-ish lines. I'll leave it up to you.

I'm OK to use the PlanFragmentIds as a sort key directly.

Reply:
The code is almost exactly the same for map vs. list, so i'd rather be explicit. Leaving the map, but changing it to be sorted so it's easier to look at.